bin/hdfcoinc/pycbc_plot_singles_timefreq

```diff
@@ -200,13 +200,22 @@ for gate_type, hatch_style in [('file', '\\'), ('auto', '/')]:
         gate_pad = trigs['gating/' + gate_type + '/pad'][:]
     except KeyError:
         continue
-    for gt, gw, gp in zip(gate_time, gate_width, gate_pad):
-        # only try to plot gates within the time window
+    gate_unique = list(frozenset(zip(gate_time, gate_width, gate_pad)))
+    for gt, gw, gp in gate_unique:
+        # only plot gates within the plot time window
         if gt + gw < opts.gps_start_time or gt - gw > opts.gps_end_time:
             continue
         ax.axvspan(gt - gw - center_time, gt + gw - center_time,
                    hatch=hatch_style, facecolor='none', edgecolor='#00ff00')
 
+if opts.veto_file:
+    logging.info('Loading and plotting veto segments')
+    veto_segs = pycbc.events.veto.select_segments_by_definer(
+            opts.veto_file, ifo=opts.detector)
+    for seg in veto_segs:
+        ax.axvspan(seg[0] - center_time, seg[1] - center_time,
```

Comment:
probably we would rather not plot every segment in a whole run, there could be >thousands. prefer to do similar to the gates and preselect the ones that would appear on the plot.